pkg/apis/eventing/struct_validation.go

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package eventing
+
+import (
+	"reflect"
+)
+
+// HasMultipleSetFields checks if the struct has more than one field with a non-zero value
+func HasMultipleSetFields(o interface{}) bool {
+	val := reflect.ValueOf(o)
+	fieldSet := false
+	for i := 0; i < val.NumField(); i++ {
+		f := val.Field(i)
+		if !IsEmptyField(f) {
+			if !fieldSet {
+				fieldSet = true
+			} else {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+// IsEmptyField checks if a struct field is empty by comparing it against its zero value
+func IsEmptyField(v reflect.Value) bool {
```

Comment:
This is only used in this package in `HasMultipleSetFields`
```suggestion
// isEmptyField checks if a struct field is empty by comparing it against its zero value
func isEmptyField(v reflect.Value) bool {
```

Reply:
ack, but do you have a reason why should we fall to unexported by default? I'm just curious